src/chatdDb.h

```diff
@@ -413,6 +428,42 @@ class ChatdSqliteDb: public chatd::DbInterface
         mDb.query("delete from history where chatid = ?", mChat.chatId());
         setHaveAllHistory(false);
     }
+
+    virtual void addAttachmentMessageToHistoryNode(const chatd::Message& msg, chatd::Idx idx)
+    {
+        if (getIdxOfMsgid(msg.id(), "node_history") == CHATD_IDX_INVALID)
+        {
+            addMessage(msg, idx, "node_history");
+        }
+    }
+
+    virtual void removeAttachmentMessageToHistoryNode(const chatd::Message& msg)
+    {
+        mDb.query("update node_history set type = ?, data = ?, updated = ?, userid = ?, is_encrypted = ? where chatid = ? and msgid = ?",
+                  msg.type, msg, msg.updated, msg.userid, msg.isEncrypted(), mChat.chatId(), msg.id());
+    }
+
+    virtual void truncateHistoryNode(karere::Id id)
+    {
+        auto idx = getIdxOfMsgid(id, "node_history");
```

Comment:
If the `id` of the truncate-message is not part of node-history, the returned index will be `CHAT_INVALID_IDX`, which is higher than any valid index, resulting on the full node-history cleared up (which is not the expected behavior).